packages/react/src/ReactStartTransition.js

```diff
@@ -6,17 +6,47 @@
  *
  * @flow
  */
+import type {StartTransitionOptions} from 'shared/ReactTypes';
 
 import ReactCurrentBatchConfig from './ReactCurrentBatchConfig';
-import {warnOnSubscriptionInsideStartTransition} from 'shared/ReactFeatureFlags';
+import {
+  warnOnSubscriptionInsideStartTransition,
+  enableTransitionTracing,
+} from 'shared/ReactFeatureFlags';
 
-export function startTransition(scope: () => void) {
+// Intentionally not named imports because Rollup would use dynamic dispatch for
+// CommonJS interop named imports.
+import * as Scheduler from 'scheduler';
+
+const {unstable_now: now} = Scheduler;
+
+export function startTransition(
+  scope: () => void,
+  options?: StartTransitionOptions,
+) {
   const prevTransition = ReactCurrentBatchConfig.transition;
   ReactCurrentBatchConfig.transition = 1;
+
+  let prevTransitionInfo = null;
+  if (enableTransitionTracing) {
+    prevTransitionInfo = ReactCurrentBatchConfig.transitionInfo;
+    if (options !== undefined && options.name !== undefined) {
+      ReactCurrentBatchConfig.transitionInfo = {
+        name: options.name,
+        startTime: -1,
+      };
+    }
```

Comment:
I _think_ this might be a logic bug?

If there's no new name, shouldn't we also (temporarily) clear out the previous `transitionInfo` before invoking the scope? or am I misreading this?